python/lsst/ts/externalscripts/base_take_ptc_flats.py

```diff
@@ -0,0 +1,451 @@
+                        f"Must be one of {self.camera.components_attr}. "
+                        f"Ignoring."
+                    )
+
+        # Ensure flats_exp_times is provided in pairs
+        if len(config.flats_exp_times) % 2 != 0:
+            raise ValueError("Flats exposure times must be provided in pairs.")
+
+        # Check that each pair in flats_exp_times contains equal values
+        for i in range(0, len(config.flats_exp_times), 2):
+            if config.flats_exp_times[i] != config.flats_exp_times[i + 1]:
+                raise ValueError(
+                    "Flats exposure times must be provided in pairs of equal values."
+                )
+
+        # Handle interleave darks settings
+        if hasattr(config, "interleave_darks"):
+            if (
+                isinstance(config.interleave_darks["dark_exp_times"], list)
+                and config.interleave_darks["n_darks"]
+            ):
+                raise ValueError(
+                    "If 'n_darks' is provided, 'dark_exp_times' must be a scalar."
+                )
+
+            if not isinstance(config.interleave_darks["dark_exp_times"], list):
+                config.interleave_darks["dark_exp_times"] = [
+                    config.interleave_darks["dark_exp_times"]
+                ] * (config.interleave_darks["n_darks"])
+            else:
+                config.interleave_darks["n_darks"] = len(
+                    config.interleave_darks["dark_exp_times"]
+                )
+
+        self.config = config
+
+        await super().configure(config)
+
+    @abc.abstractmethod
+    def get_instrument_name(self):
+        """Abstract method to be defined in subclasses to provide the
+        instrument name.
+        """
+        raise NotImplementedError()
+
+    @abc.abstractmethod
+    def get_instrument_configuration(self) -> dict:
+        """Abstract method to get the instrument configuration.
+
+        Returns
+        -------
+        dict
+            Dictionary with instrument configuration.
+        """
+        raise NotImplementedError()
+
+    @abc.abstractmethod
+    def get_instrument_filter(self) -> str:
+        """Abstract method to get the instrument filter configuration.
+
+        Returns
+        -------
+        str
+            Instrument filter configuration.
+        """
+        raise NotImplementedError()
+
+    async def setup_electrometer(
+        self, mode: str, range: float, integration_time: float
+    ) -> None:
+        """Setup the Electrometer with specified mode, range,
+        and integration time.
+
+        Parameters
+        ----------
+        mode : `str`
+            Electrometer measurement mode.
+        range : `float`
+            Electrometer measurement range. -1 for autorange.
+        integration_time : `float`
+            Electrometer integration time.
+        """
+        try:
+            electrometer_mode = getattr(UnitToRead, mode).value
+        except AttributeError:
```

Comment:
Why convert an `AttributeError` into a `ValueError`. If anything (which I don't think is necessary) convert it into a `RuntimeError`... But I would just not this at all. Just do 

```
electrometer_mode = getattr(UnitToRead, mode).value
```